tmt/base.py

```diff
@@ -2130,6 +2130,18 @@ def lint_unknown_keys(self) -> LinterReturn:
 
         yield LinterOutcome.PASS, 'correct keys are used'
 
+    def lint_empty_env_files(self) -> LinterReturn:
+        """ P001: all files are not empty """
+
+        env_file = self.plan_environment_file
+
+        if (self.my_run and env_file.exists() and env_file.stat().st_size == 0):
```

Comment:
I'm not sure this would work: by design, `lint_*` callbacks are called not in runtime, `tmt run ...`, but e.g. before committing changes, via `tmt * lint`. Therefore `self.my_run` should always be unset when linters run.

I think there is also a misunderstanding in your way: `plan_environment_file` is a per-plan file, provided by tmt to plans and tests, with path exposed via `TMT_PLAN_ENVIRONMENT_FILE` envvar, and as such does exist only in the context of `tmt run`. Which makes it non-lintable entity.

If you wanted to check whether files referenced by plan's `environment-file`, https://tmt.readthedocs.io/en/stable/spec/plans.html#environment-file, you'd need to check `self.node`, `self.node.get("environment-file") or []` should give you the list of files referenced by the plan.

Reply:
@happz thanks for the feedback I am still a newbie here, just a couple of comments. The idea of this patch is to cover the issue #2872 which mentions to report the fail in the `tmt lint` so that should be the idea right?, or maybe you want to rethink the intention of the issue. Regarding the plan environment file I changed the code according to your review, I hope now it's appropiate
cc @psss

Reply:
I think the idea matches the issue, you just focused on the wrong "environment file", that's all :)